tests/helpers/test_config_validation.py

```diff
@@ -481,26 +481,71 @@ def test_datetime():
     schema('2016-11-23T18:59:08')
 
 
-def test_deprecated(caplog):
-    """Test deprecation log."""
+def test_deprecated_logging(caplog):
+    """Test deprecation logs the correct message in each situation."""
     schema = vol.Schema({
         'venus': cv.boolean,
-        'mars': cv.boolean
+        'mars': cv.boolean,
+        'jupiter': cv.boolean
     })
+
     deprecated_schema = vol.All(
         cv.deprecated('mars'),
         schema
     )
 
+    deprecated_schema_with_invalidation_version = vol.All(
```

Comment:
Debating splitting each of these out into their own test case. Benefits of doing so is that we can also check the behavior of moving the value to `replacement_key` inline as well as the `invalidation_version` logic rather than having to write separate tests to test that.